src/main/java/mil/dds/anet/resources/ReportResource.java

```diff
@@ -374,7 +376,19 @@ public Report submitReport(@GraphQLRootContext Map<String, Object> context,
     logger.debug("Attempting to submit report {}, which has advisor org {} and primary advisor {}",
         r, r.getAdvisorOrg(), r.getPrimaryAdvisor());
 
-    // TODO: this needs to be done by either the Author, a Superuser for the AO, or an Administrator
+    if (!r.getAuthor().getUuid().equals(user.getUuid())
+        && !AuthUtils.isSuperUserForOrg(user, r.getAdvisorOrgUuid(), true)
+        && !AuthUtils.isAdmin(user)) {
+      throw new WebApplicationException(
+          "Cannot submit report unless the report's author, his/her super user or an admin",
```

Comment:
Suggested change: `"Cannot submit report unless you are the report's author, his/her super user or an admin"`